validation/note.js

```diff
@@ -0,0 +1,40 @@
+import isEmpty from "lodash.isempty";
+import { isBlank, isValidModelId } from "./validationUtils";
+import Note from "../models/note";
+
+export const validateNoteUpdate = (noteId, text) => {
+  let errors = {};
+  if (isBlank(text)) {
+    errors.note = "Note cant be empty";
+  }
+  return {
+    errors,
+    hasErrors: !isEmpty(errors),
+  };
+};
+
+export const validateNoteCreate = (text) => {
+  let errors = {};
+  if (isBlank(text)) {
+    errors.note = "Note can't be empty";
+  }
+  return {
+    errors,
+    hasErrors: !isEmpty(errors),
+  };
+};
+
+export const validateNoteDomainMiddleware = (req, res, next) => {
```

Comment:
Usually we keep our middlewares in routes files. It better to keep it here but to be consistent better move it to routes file. Or move all other middlewares to validation files.